builder/comp-builder.nix

```diff
@@ -250,6 +271,35 @@ in stdenv.mkDerivation ({
     runHook postCheck
   '';
 
+  haddockPhase = ''
+    runHook preHaddock
```

Comment:
Shouldn't the hooks be inside the `optionalString doHaddock`?

Reply:
I assumed Peti had a good reason for doing it like that in the nixpkgs haskell builder.
But actually it's more consistent with the other phases (e.g. `checkPhase`) to not run the hooks.
Fixed.